Sources/DistributedActors/Cluster/SWIM/SWIMActorShell.swift

```diff
@@ -15,9 +15,517 @@
+        sequenceNumber: SWIM.SequenceNumber
+    ) async -> SWIM.PingResponse {
+        await withCheckedContinuation { continuation in
+            self.log.trace("Received ping@\(sequenceNumber)", metadata: self.swim.metadata([
+                "swim/ping/origin": "\(origin.id)",
+                "swim/ping/payload": "\(payload)",
+                "swim/ping/seqNr": "\(sequenceNumber)",
+            ]))
+
+            self.swim.onPing(
+                pingOrigin: origin,
+                payload: payload,
+                sequenceNumber: sequenceNumber
+            ).forEach { directive in
+                switch directive {
+                case .gossipProcessed(let gossipDirective):
+                    self.handleGossipPayloadProcessedDirective(gossipDirective)
+
+                case .sendAck(_, let pingedTarget, let incarnation, let payload, let sequenceNumber):
+                    continuation.resume(returning: .ack(target: pingedTarget, incarnation: incarnation, payload: payload, sequenceNumber: sequenceNumber))
+                }
+            }
+        }
+    }
+    
+    distributed func pingRequest(
+        target: SWIMActorShell,
+        pingRequestOrigin: SWIMActorShell,
+        payload: SWIM.GossipPayload,
+        sequenceNumber pingRequestSequenceNumber: SWIM.SequenceNumber
+    ) async -> SWIM.PingResponse {
+        await withCheckedContinuation { continuation in
+            self.log.trace("Received pingRequest@\(pingRequestSequenceNumber) [\(target)] from [\(pingRequestOrigin)]", metadata: self.swim.metadata([
+                "swim/pingRequest/origin": "\(pingRequestOrigin)",
+                "swim/pingRequest/payload": "\(payload)",
+                "swim/pingRequest/seqNr": "\(pingRequestSequenceNumber)",
+            ]))
+
+            self.swim.onPingRequest(
+                target: target,
+                pingRequestOrigin: pingRequestOrigin,
+                payload: payload,
+                sequenceNumber: pingRequestSequenceNumber
+            ).forEach { directive in
+                switch directive {
+                case .gossipProcessed(let gossipDirective):
+                    self.handleGossipPayloadProcessedDirective(gossipDirective)
+
+                case .sendPing(let target, let payload, let pingRequestOrigin, let pingRequestSequenceNumber, let timeout, let pingSequenceNumber):
+                    Task {
+                        let response = await self.sendPing(
+                            to: target,
+                            payload: payload,
+                            pingRequestOrigin: pingRequestOrigin,
+                            pingRequestSequenceNumber: pingRequestSequenceNumber,
+                            timeout: timeout,
+                            sequenceNumber: pingSequenceNumber
+                        )
+                        continuation.resume(returning: response)
+                    }
+                }
+            }
+        }
```

Comment:
Turns out using the swim instance is hard to use with async await heh I guess :-(